databases/week3/Queries.sql

```diff
@@ -0,0 +1,158 @@
+DELETE FROM meal
+WHERE price = 100;
+
+-- Reservation
+# Get all reservations
+SELECT *
+FROM reservation;
+
+# Add a new reservation
+INSERT INTO reservation (number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email) 
+SELECT 2, id, now(), 71708085, 'Aarika Ellingworth', 'aellingworth0@harvard.edu' 
+FROM meal
+WHERE title = 'Homemade Pasta';
+
+INSERT INTO reservation (number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email) 
+SELECT 3, id, now(), 71706024, 'John Dark', 'john.dark@gmail.com' 
+FROM meal
+WHERE title = 'Homemade Pasta';
+
+INSERT INTO reservation (number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email) 
+SELECT 5, id, now(), 71708085, 'Aarika Ellingworth', 'aellingworth0@harvard.edu' 
+FROM meal
+WHERE title = 'Homemade spagetti';
+
+INSERT INTO reservation (number_of_guests, meal_id, created_date, contact_phonenumber, contact_name, contact_email) 
+SELECT 5, id, now(), 71706024, 'John Dark', 'john.dark@gmail.com' 
+FROM meal
+WHERE title = 'Homemade spagetti';
+
+# Get a reservation with any id, fx 1
+SELECT *
+FROM reservation
+WHERE id = 1;
+
+# Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE reservation
+SET number_of_guests = 5, contact_name = 'Angelia Jully'
+WHERE id = 1;
+
+#Delete a reservation with any id, fx 1
+DELETE FROM reservation
+WHERE id = 1;
+
+-- Review
+INSERT INTO review (title, description,meal_id, stars, created_date)
+SELECT 'deleicious', 'pesto was super fresh', id, 4, now()
+FROM meal
+WHERE title = 'Homemade Pasta';
+
+INSERT INTO review (title, description,meal_id, stars, created_date)
+SELECT 'Good', 'Thegarlic sauce was the best', id, 4, now()
+FROM meal
+WHERE title='Homemade spagetti';
+
+INSERT INTO review (title, description,meal_id, stars, created_date)
+SELECT 'Simply wow', 'Gnocchi with pesto, ragu or mushroom sauce was unique.', id, 5, now()
+FROM meal
+WHERE title='Homemade Gnocchi';
+
+# Get all reviews
+SELECT *
+FROM review;
+
+# Add a new review
+INSERT INTO review (title, description, meal_id, stars, created_date)
+SELECT 'Best soup ever', 'The taste of soup was rich.', id, 5, now()
+FROM meal
+WHERE title='Authentic Tortellini Soup';
+
+# Get a review with any id, fx 1
+SELECT review.*
+FROM review
+WHERE id = 17;
+
+# Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE review
+SET title = 'updated review', description = 'The light and garlicky spaghetti sauce was lovely.'
+WHERE id = 17;
+
+# Delete a review with any id, fx 1
+DELETE FROM review
+WHERE id = 15;
+
+-- Additional queries
+# Get meals that has a price smaller than a specific price fx 90
+SELECT meal.title, meal.price
+FROM meal
+WHERE PRICE < 150;
+
+# Get meals that still has available reservations
+SELECT meal.title, (meal.max_reservations - IFNULL(SUM(reservation.number_of_guests), 0)) AS available_reservations
```

Comment:
Consider using COALESCE() here instead of IFNULL():  https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#function_coalesce

The reasoning is that COALESCE() is part of the SQL standard and IFNULL() is not, so if you use another DBMS (e.g. postgresql or sqlite) COALESCE() will be available but IFNULL() might not be.